test/suite/test_util18.py

```diff
@@ -0,0 +1,128 @@
+                self.check_file_not_contains(log_file, '"value": "%s\\u0000"' % val)
+
+            # Convert our KEY/VAL strings to their expected hex value
+            hex_key = codecs.encode(key.encode(), 'hex')
+            val_key = codecs.encode(val.encode(), 'hex')
+            # Check if the KEY/VAL commits exist in the log file (in hex form)
+            if expect_keyval_hex:
+                self.check_file_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+            else:
+                self.check_file_not_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_not_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+
+    def test_printlog_file(self):
+        """
+        Test printlog on a populated table
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog"], outfilename='printlog.out')
+        self.check_non_empty_file('printlog.out')
+        self.check_populated_printlog('printlog.out', True, False)
+
+    def test_printlog_hex_file(self):
+        """
+        Test printlog with hexadecimal formatting on a populated table
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog", "-x"], outfilename='printlog-hex.out')
+        self.check_non_empty_file('printlog-hex.out')
+        self.check_populated_printlog('printlog-hex.out', True, True)
+
+    def test_printlog_message(self):
+        """
+        Test printlog with messages-only formatting on a populated table
+        """
+        # Run reconfigure with compatibility to force 'COMPATIBILITY' message
+        self.conn.reconfigure('compatibility=(release=3.2.0)')
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog", "-m"], outfilename='printlog-message.out')
+        self.check_non_empty_file('printlog-message.out')
+        self.check_file_contains('printlog-message.out', 'COMPATIBILITY: Version now 3')
+        self.check_populated_printlog('printlog-message.out', False, False)
+
+    def test_printlog_lsn_offset(self):
+        """
+        Test printlog with an LSN offset provided
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog", '-l 1,128,1,128'], outfilename='printlog-lsn-offset.out')
+        self.check_file_contains('printlog-lsn-offset.out', '"lsn" : [1,128]')
+        self.check_file_not_contains('printlog-lsn-offset.out', '"lsn" : [1,256]')
+        self.check_populated_printlog('printlog-lsn-offset.out', False, False)
+        self.runWt(["printlog", '-l 1,128'], outfilename='printlog-lsn-offset.out')
```

Comment:
I believe "-l 1,0" also means start at the beginning of log file 1. There are some configurations where the first record is not at offset 128. But that only works with this situation where you give only a starting LSN. It might be interesting to make sure that `-l 1,128` (or `-l first_lsn` if you take the suggestion above) and -l 1,0` give the same output.

Reply:
Rereading this, it may not be clear. Let me know if you need clarification here.

Reply:
That's really good to know. I added a test to compare `-l 1,0` to `-l first_lsn` in the top commit (`5fec49b`). Hopefully I understood the idea correctly.